### PR TITLE
Add Softplus activation function

### PR DESCRIPTION
**Pull request by ivmarkp**
All tests pass locally. Please review.

Now, working on adding ELU layer. Will open a PR ASAP.

### CODE REVIEW

File: src/mlpack/methods/ann/activation_functions/softplus_function.hpp [hunk 1]
```diff
@@ -0,0 +1,114 @@
+/**
+ * @file softplus_function.hpp
+ * @author Vivek Pal
+ *
+ * Definition and implementation of the softplus function.
+ *
```
Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: Can you cite the "Incorporating Second-Order Functional Knowledge for Better Option Pricing" by C. Dugas et al. paper that introduced the softplus function? Take a look at the softsign function header for an example.

Comment (ivmarkp) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: Sure. I found the citation [here](http://www.iro.umontreal.ca/~lisa/publications2/index.php/export/publication/67/bibtex). Is that format correct? I think may be "abstract" field can be dropped from it for our purposes.

Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: Sounds good, you can also drop url, topics and cat.

Comment (ivmarkp) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: Okay, pushing the commit in a minute.

Comment (zoq) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: The required fields of INPROCEEDINGS are: author, title, booktitle, year so, we should make sure there is a booktitle.

Comment (ivmarkp) [inline src/mlpack/methods/ann/activation_functions/softplus_function.hpp]: Yes, sorry I missed that. Booktitle is "Advances in Neural Information Processing Systems".

**Comment by zoq** [comment]
Thanks again for the great contribution, looking forward to take a look at the other PR as well.

**Comment by ivmarkp** [comment]
Thanks, will address your review comments on the other PR as soon as I can.

**Pull request merged**